sdk/communication/azure-communication-callautomation/src/main/java/com/azure/communication/callautomation/CallDialogAsync.java

```diff
@@ -0,0 +1,117 @@
+     * @param callDialogInternal Internal implementation of CallDialog
+     */
+    public CallDialogAsync(String callConnectionId, CallDialogsImpl callDialogInternal) {
+        this.callConnectionId = callConnectionId;
+        this.dialogsInternal = callDialogInternal;
+        this.logger = new ClientLogger(CallDialogAsync.class);
+    }
+
+    /**
+     * Start Dialog
+     *
+     * @param startDialogOptions startDialog options
+     * @return Response for successful startDialog request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DialogStateResult>> startDialog(StartDialogOptions startDialogOptions) {
+        return startDialogWithResponse(startDialogOptions, null);
+    }
+
+    /**
+     * Start Dialog
+     *
+     * @param startDialogOptions startDialog options
+     * @param operationContext operationContext (pass null if not applicable)
+     * @return Response for successful startDialog request.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<DialogStateResult>> startDialogWithResponse(StartDialogOptions startDialogOptions, String operationContext) {
+        return withContext(context -> startDialogWithResponseInternal(startDialogOptions, operationContext, context));
+    }
+
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    Mono<Response<DialogStateResult>> startDialogWithResponseInternal(StartDialogOptions startDialogOptions, String operationContext, Context context) {
+        try {
+            context = context == null ? Context.NONE : context;
+
+            DialogOptions dialogOptions = new DialogOptions();
+            dialogOptions.setDialogContext(startDialogOptions.getDialogContext());
+            dialogOptions.setBotAppId(startDialogOptions.getBotId());
+
+            StartDialogRequestInternal requestInternal = new StartDialogRequestInternal()
+                .setDialogOptions(dialogOptions)
+                .setOperationContext(operationContext);
+
+            return dialogsInternal.startDialogWithResponseAsync(callConnectionId, startDialogOptions.getDialogId(), requestInternal, context).
+                map(response -> new SimpleResponse<>(response, DialogStateResponseConstructorProxy.create(response.getValue())));
+        } catch (RuntimeException e) {
+            return monoError(logger, e);
+        }
+    }
+
+    /**
+     * Start Dialog
+     *
+     * @param dialogId startDialog options
+     * @return Response for successful startDialog request.
+     */
+    public Mono<Void> stopDialog(String dialogId) {
+        return stopDialogWithResponse(dialogId, null).then();
+    }
+
+    /**
+     * Start Dialog
+     *
+     * @param dialogId startDialog options
+     * @param operationContext operationContext (pass null if not applicable)
+     * @return Response for successful startDialog request.
+     */
+    public Mono<Response<Void>> stopDialogWithResponse(String dialogId, String operationContext) {
```

Comment:
It seems this stopDialogWithResponse async method have operationContext, but sync stopDialogWithResponse doesnt. I am wondering if this is intended?